### PR TITLE
Allow querying other dnsclasses besides IN (CH, ..)

### PR DESCRIPTION
**Pull request by rytilahti**
I can clean this up however you prefer, I just want to put it up before forgetting about it.

* Add optional dnsclass parameter to query functions
* Non-API breaking change, falls back to IN íf no class is given
* Allow TXT records on CHAOS

### CODE REVIEW

File: deps/c-ares/src/ares_parse_txt_reply.c [hunk 1]
```diff
@@ -114,7 +114,7 @@ ares__parse_txt_reply (const unsigned char *abuf, int alen,
         }
 
       /* Check if we are really looking at a TXT record */
-      if (rr_class == C_IN && rr_type == T_TXT)
+      if ((rr_class == C_IN || rr_class == C_CHAOS) && rr_type == T_TXT)
```
Comment (saghul) [inline deps/c-ares/src/ares_parse_txt_reply.c:117]: Is this change a port from upstream?

Comment (rytilahti) [reply deps/c-ares/src/ares_parse_txt_reply.c:117]: No, that's not from upstream, so I suppose it's better if I submit it there, too.

Other than that, I'll add some tests and update this PR.

File: src/pycares/__init__.py [hunk 2]
```diff
@@ -501,22 +501,32 @@ def gethostbyname(self, name, family, callback):
...
 
+        DNSCLASS_MAP = {
+            "IN": _lib.C_IN,
+            "CH": _lib.C_CHAOS,
+            "HS": _lib.C_HS,
+            "NONE": _lib.C_NONE,
+            "ANY": _lib.C_ANY,
+        }
+
+        dnsclass_cares = DNSCLASS_MAP.get(dnsclass, _lib.C_IN)
```
Comment (rytilahti) [inline src/pycares/__init__.py:525]: Should this mapping be elevated to the top level (akin to `QUERY_TYPE_x` symbols)?

Comment (saghul) [reply src/pycares/__init__.py:525]: yeah

File: deps/c-ares/src/ares_parse_txt_reply.c (hunk 1 above)
Comment (rytilahti) [reply deps/c-ares/src/ares_parse_txt_reply.c:117]: Upstream PR got merged.

File: src/pycares/__init__.py (hunk 2 above)
Comment (rytilahti) [reply src/pycares/__init__.py]: Done.

File: src/pycares/__init__.py [hunk 3]
```diff
@@ -65,6 +65,13 @@
...
+
+    # Query classes
+    "QUERY_CLASS_IN": "C_IN",
+    "QUERY_CLASS_CH": "C_CHAOS",
+    "QUERY_CLASS_HS": "C_HS",
+    "QUERY_CLASS_NONE":"C_NONE",
+    "QUERY_CLASS_ANY": "C_ANY",
 }
 
 for k, v in exported_pycares_symbols_map.items():
```
Comment (rytilahti) [inline src/pycares/__init__.py:77]: Btw, it could probably be worth to just type these out, if not only to allow auto-completion for IDEs.

**Pull request merged**